src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1660,14 +1807,19 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode)
 			newTargetEntry->resjunk = false;
 			newTargetEntry->resno = targetProjectionNumber;
 			targetProjectionNumber++;
-
+			targetIndex++;
 			newTargetEntryList = lappend(newTargetEntryList, newTargetEntry);
 		}
 	}
 
+	if (groupClauseList != NIL)
+	{
+		originalGroupClauseList = list_concat(originalGroupClauseList, groupClauseList);
```

Comment:
originalGroupClauseList was not necessary. Using groupClauseList and newGroupClauseList has the same effect.